sw/airborne/modules/computer_vision/opticflow_module.c

```diff
@@ -151,14 +151,12 @@ struct image_t *opticflow_module_calc(struct image_t *img)
 
   // Do the optical flow calculation
   static struct opticflow_result_t temp_result; // static so that the number of corners is kept between frames
-  bool flow_successful = opticflow_calc_frame(&opticflow, img, &temp_result);
-
-  // Copy the result if finished
-  pthread_mutex_lock(&opticflow_mutex);
-  opticflow_result = temp_result;
-  opticflow_got_result = flow_successful;
-
-  // release the mutex as we are done with editing the opticflow result
-  pthread_mutex_unlock(&opticflow_mutex);
+  if(opticflow_calc_frame(&opticflow, img, &temp_result)){
+    // Copy the result if finished
+    pthread_mutex_lock(&opticflow_mutex);
+    memcpy(&opticflow_result, &temp_result, sizeof(struct opticflow_result_t));
```

Comment:
True, the compiler will likely use a memcpy behind the scenes for this. Here I am trying to highlight what is happening here so that if the structure changes in the future to contain more complex variables (like arrays of arrays) that something more is required here. If it is less clear I can revert.

Reply:
I think the assignment is more concise (and therefore better), in both cases it should be clear what is happening as they both mean the same. I think the memcpy might make it harder to spot such problems in the future, as it looks like someone has already thought about this. If you're really worried, it might be better to add a comment that this is only a shallow copy.